models/articles.model.js

```diff
@@ -16,14 +32,30 @@ function selectArticles() {
     FROM articles
     LEFT JOIN comments ON articles.article_id = comments.article_id
     GROUP BY articles.article_id
-    ORDER BY articles.created_at DESC;
-    `
-    )
-    .then(({ rows }) => {
-      return rows;
-    });
+    ORDER BY articles.${sortBy} ${order}
+    `;
+
+  return db.query(queryString).then(({ rows }) => {
+    return rows;
+  });
 }
 
+function checkValidQueryParams(requestQuery) {
+  return new Promise((resolve, reject) => {
+    const validQueryParams = ["sort_by", "order"];
+    const queryParams = Object.keys(requestQuery);
+
+    const invalidParams = queryParams.filter((param) => {
+      return !validQueryParams.includes(param);
+    });
+
+    if (invalidParams.length > 0) {
+      reject({ status: 400, msg: "Invalid query" });
+    } else {
+      resolve();
+    }
+  });
+}
```

Comment:
This looks like the product of a decent amount of hard work, but I am sorry to say I don't think any of it is necessary to have in your endpoint